### PR TITLE
feat(HFX-1987): add top left action to action modal

### PR DESCRIPTION
**Pull request by MariannaPrint**
## Pull Request Details

**Jira ticket**: [HFX-1987](https://newcross.atlassian.net/browse/HFX-1897)

### Description
Add an optional prop for having a top right action on the action modal , instead of an x action
image

### Contributors

- [ ] Changes covered by unit tests
- [ ] Tests check for boundary conditions, e.g. just below, at, and just above a boundary, zero, negative numbers, NaN
- [ ] Storybook is updated for new & existing components
- [ ] Showcase the changes to UI/UX team
- [ ] The project documentation has been updated, as appropriate. `README.md` for details, Confluence for the "Big Picture"

### Code Owners

- [ ] Open storybook to verify the changes
- [ ] Ensure that changes are always backward compatible
- [ ] Ensure commit message follows conventional commit format

### CODE REVIEW

File: packages/react/src/components/ActionModal/ActionModal.types.ts [hunk 1]
```diff
@@ -76,4 +78,9 @@ export type ActionModalPropsStrict = {
    * Default is false.
    */
   shouldHideIcon?: boolean;
+  /**
+   * Prop to control the CTA text to be shown in the top right corner of the header.
+   * Default is null.
+   */
+  topRightAction?: ReactElement<LinkProps> | string;
```
Comment (Vangelis-Vlassopoulos) [inline packages/react/src/components/ActionModal/ActionModal.types.ts:85]: In types "Strict" we should have non-optional values. Could you please omit the "?" ?

File: packages/react/src/components/ActionModal/ActionModal.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@ import Container from '../Container';
 import Icon from '../Icon';
 import { useIsBottomSheetBreakpoint } from './hooks/useIsBottomSheetBreakpoint';
 import { isValidElement, cloneElement } from 'react';
+import { AlertAction } from '../Alert/AlertAction';
 
 export type ActionModalProps = OptionalProps<
   ActionModalPropsStrict,
```
[the remark this answers is not in the document]
Comment (Vangelis-Vlassopoulos) [reply packages/react/src/components/ActionModal/ActionModal.tsx:23]: also add the default value of your new prop in the normalizeActionModalProps function!

File: packages/react/src/components/ActionModal/ActionModal.tsx [hunk 3]
```diff
@@ -113,15 +115,23 @@ const ActionModal = (_props: ActionModalProps) => {
...
-                    icon={faXmark}
-                    color="defaultDark"
-                    onClick={onDismiss}
-                    testID={`${baseName}close-icon`}
-                  />
-                )}
+                {!canCloseOnActionOnly &&
+                  !isBottomSheet &&
+                  (topRightAction ? (
+                    <AlertAction
```
Comment (Vangelis-Vlassopoulos) [inline packages/react/src/components/ActionModal/ActionModal.tsx:121]: Why don't we opt for a `Link` here?
